File: pset6/credit/credit.py
# Made by PMBC
number = input("Number: ")

# AMEX
if len(number) == 15 and number[0] == '3' and (number[1] == '4' or number[1] == '7'):
    print('AMEX')
    
# VISA
elif len(number) == 13 and number[0] == '4':
    print('VISA')
    
# MASTERCARD
elif len(number) == 16:
    if number[0] == '5' and (int(number[1]) < 6 and int(number[1]) > 0):
        print('MASTERCARD')
    elif number[0] == '4':
        print('VISA')
    else:
        print('INVALID')
    
# INVALID
else:
    print("INVALID")

File: pset1/credit/credit.c
// Made By PMBC
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    long card = get_long("Number: ");
    string card_type = "INVALID";
    int t = 0;
    for(int i=15;i>=0;i--)
    {
        long n=1;
        for(int x=0;x<i;x++)
        {
            n = n*10;
        }
        int dig = card/n;
        card = card%n;
        if (i%2==1){if(dig*2>9){t+=1;t+=(dig*2)%10;}else{t+=dig*2;}}
        else{t+=dig;}
        if (i==15 && dig==5)
        {
            long n2=n/10;
            if(card/n2>0 && card/n2<6){card_type="MASTERCARD";}
        }
        else if (i==15 && dig == 4){card_type="VISA";}
        else if (i==12 && dig == 4){card_type="VISA";}
        else if (i==14 && dig == 3)
        {
            long n2=n/10;
            if(card/n2==4 || card/n2==7){card_type="AMEX";}
        }
    }
    if (t%10!=0){card_type="INVALID";}
    printf("%s\n",card_type);
}
